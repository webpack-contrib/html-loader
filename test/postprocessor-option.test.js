import {
  compile,
  execute,
  getCompiler,
  getErrors,
  getModuleSource,
  getWarnings,
  readAsset,
} from "./helpers";

describe("'postprocess' option", () => {
  it('should work with the "postprocessor" option', async () => {
    const compiler = getCompiler("postprocessor.html", {
      postprocessor: (content, loaderContext) => {
        expect(typeof content).toBe("string");
        expect(loaderContext).toBeDefined();

        return content.replace(/<%=/g, '" +').replace(/%>/g, '+ "');
      },
    });
    const stats = await compile(compiler);

    expect(getModuleSource("./postprocessor.html", stats)).toMatchSnapshot(
      "module",
    );
    expect(
      execute(readAsset("main.bundle.js", compiler, stats)),
    ).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should work with async "postprocessor" function option', async () => {
    const compiler = getCompiler("preprocessor.hbs", {
      postprocessor: async (content, loaderContext) => {
        await expect(typeof content).toBe("string");
        await expect(loaderContext).toBeDefined();

        return content.replace(/<%=/g, '" +').replace(/%>/g, '+ "');
      },
    });
    const stats = await compile(compiler);

    expect(getModuleSource("./preprocessor.hbs", stats)).toMatchSnapshot(
      "module",
    );
    expect(
      execute(readAsset("main.bundle.js", compiler, stats)),
    ).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
