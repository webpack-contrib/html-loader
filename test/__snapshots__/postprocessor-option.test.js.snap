// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'postprocess' option should work with async "postprocessor" function option: errors 1`] = `[]`;

exports[`'postprocess' option should work with async "postprocessor" function option: module 1`] = `
"// Imports
var ___HTML_LOADER_IMPORT_0___ = new URL("./image.png", import.meta.url);
// Module
var code = \`<div>
  <p>{{firstname}} {{lastname}}</p>
  <img src="\${___HTML_LOADER_IMPORT_0___}" alt="alt" />
<div>
\`;
// Exports
export default code;"
`;

exports[`'postprocess' option should work with async "postprocessor" function option: result 1`] = `
"<div>
  <p>{{firstname}} {{lastname}}</p>
  <img src="replaced_file_protocol_/webpack/public/path/image.png" alt="alt" />
<div>
"
`;

exports[`'postprocess' option should work with async "postprocessor" function option: warnings 1`] = `[]`;

exports[`'postprocess' option should work with the "postprocessor" option #1: errors 1`] = `[]`;

exports[`'postprocess' option should work with the "postprocessor" option #1: module 1`] = `
"// Imports
var ___HTML_LOADER_IMPORT_0___ = new URL("./image.png", import.meta.url);
// Module
var code = "<img src=\\"" + ___HTML_LOADER_IMPORT_0___ + "\\">\\n<img src=\\"" + 'Hello ' + (1+1) + "\\">\\n<img src=\\"" + require('./image.png') + "\\">\\n<img src=\\"" + new URL('./image.png', import.meta.url) + "\\">\\n<div>" + require('./gallery.html').default + "</div>\\n<!--Works fine, but need improve testing <div>< %= (await import('./gallery.html')).default % ></div>-->\\n";
// Exports
export default code;"
`;

exports[`'postprocess' option should work with the "postprocessor" option #1: result 1`] = `
"<img src="replaced_file_protocol_/webpack/public/path/image.png">
<img src="Hello 2">
<img src="/webpack/public/path/image.png">
<img src="replaced_file_protocol_/webpack/public/path/image.png">
<div><h2>Gallery</h2></div>
<!--Works fine, but need improve testing <div>< %= (await import('./gallery.html')).default % ></div>-->
"
`;

exports[`'postprocess' option should work with the "postprocessor" option #1: warnings 1`] = `[]`;

exports[`'postprocess' option should work with the "postprocessor" option: errors 1`] = `[]`;

exports[`'postprocess' option should work with the "postprocessor" option: module 1`] = `
"// Imports
var ___HTML_LOADER_IMPORT_0___ = new URL("./image.png", import.meta.url);
// Module
var code = \`<img src="\${___HTML_LOADER_IMPORT_0___}">
<img src="\${ 'Hello ' + (1+1) }">
<img src="\${ require('./image.png') }">
<img src="\${ new URL('./image.png', import.meta.url) }">
<div>\${ require('./gallery.html').default }</div>
<!--Works fine, but need improve testing <div>< %= (await import('./gallery.html')).default % ></div>-->
\`;
// Exports
export default code;"
`;

exports[`'postprocess' option should work with the "postprocessor" option: result 1`] = `
"<img src="replaced_file_protocol_/webpack/public/path/image.png">
<img src="Hello 2">
<img src="/webpack/public/path/image.png">
<img src="replaced_file_protocol_/webpack/public/path/image.png">
<div><h2>Gallery</h2></div>
<!--Works fine, but need improve testing <div>< %= (await import('./gallery.html')).default % ></div>-->
"
`;

exports[`'postprocess' option should work with the "postprocessor" option: warnings 1`] = `[]`;
